gax/src/main/java/com/google/api/gax/batching/BatchingSettings.java

```diff
@@ -174,6 +174,21 @@ public BatchingSettings build() {
           settings.getDelayThreshold() == null
               || settings.getDelayThreshold().compareTo(Duration.ZERO) > 0,
           "delayThreshold must be either unset or positive");
+      if (settings.getFlowControlSettings().getLimitExceededBehavior()
```

Comment:
Regardless of how the `flowController vs flowControllerSettings` issue gets resolved, please make sure that this complex logic exists in only one place (now there are two: here and in BatcherImpl constructor) and reused if needed.

Reply:
Removed the checks in BatchingSettings and only kept them in BatchImpl. I guess it really depends on the Batcher how these settings are used, so it makes more sense to enforce the checks there?

Reply:
Sounds reasonable